File: software/examples/ruby/example_simple.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_gps'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your GPS Bricklet

ipcon = IPConnection.new # Create IP connection
gps = BrickletGPS.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get current coordinates as [latitude, ns, longitude, ew, pdop, hdop, vdop, epe]
coordinates = gps.get_coordinates

puts "Latitude: #{coordinates[0]/1000000.0} °"
puts "N/S: #{coordinates[1]}"
puts "Longitude: #{coordinates[2]/1000000.0} °"
puts "E/W: #{coordinates[3]}"

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/perl/example_simple.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletGPS;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your GPS Bricklet

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $gps = Tinkerforge::BrickletGPS->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Get current coordinates
my ($latitude, $ns, $longitude, $ew, $pdop, $hdop, $vdop, $epe) = $gps->get_coordinates();

print "Latitude: " . $latitude/1000000.0 . " °\n";
print "N/S: $ns\n";
print "Longitude: " . $longitude/1000000.0 . " °\n";
print "E/W: $ew\n";

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();
